Covid19Radar/Covid19Radar/Repository/UserDataRepository.cs

```diff
@@ -28,31 +31,157 @@ public interface IUserDataRepository
+            _loggerService = loggerService;
+        }
+
+        public Task<long> GetLastProcessDiagnosisKeyTimestampAsync(string region)
+        {
+            _loggerService.StartMethod();
+
+            try
+            {
+                var result = 0L;
+
+                var jsonString = _preferencesService.GetValue<string>(PreferenceKey.LastProcessTekTimestamp, null);
+                if (!string.IsNullOrEmpty(jsonString))
+                {
+                    var dict = JsonConvert.DeserializeObject<Dictionary<string, long>>(jsonString);
+                    if (dict.ContainsKey(region))
+                    {
+                        result = dict[region];
+                    }
+                }
+
+                return Task.FromResult(result);
+            }
+            finally
+            {
+                _loggerService.EndMethod();
+            }
+        }
+
+        public Task SetLastProcessDiagnosisKeyTimestampAsync(string region, long timestamp)
+        {
+            _loggerService.StartMethod();
+
+            try
+            {
+                var jsonString = _preferencesService.GetValue<string>(PreferenceKey.LastProcessTekTimestamp, null);
+
+                Dictionary<string, long> newDict;
+                if (!string.IsNullOrEmpty(jsonString))
+                {
+                    newDict = JsonConvert.DeserializeObject<Dictionary<string, long>>(jsonString);
+                }
+                else
+                {
+                    newDict = new Dictionary<string, long>();
+                }
+                newDict[region] = timestamp;
+                _preferencesService.SetValue(PreferenceKey.LastProcessTekTimestamp, JsonConvert.SerializeObject(newDict));
+
+                return Task.CompletedTask;
+            }
+            finally
+            {
+                _loggerService.EndMethod();
+            }
+        }
+
+        public Task RemoveLastProcessDiagnosisKeyTimestampAsync()
+        {
+            _loggerService.StartMethod();
+
+            try
+            {
+                _preferencesService.RemoveValue(PreferenceKey.LastProcessTekTimestamp);
+                return Task.CompletedTask;
+            }
+            finally
+            {
+                _loggerService.EndMethod();
+            }
+        }
+
+        #region ExposureWindow mode
+        public async Task SetExposureDataAsync(
+            List<DailySummary> dailySummaryList,
+            List<ExposureWindow> exposueWindowList
+            )
+        {
+            _loggerService.StartMethod();
+
+            dailySummaryList.AddRange(dailySummaryList);
+            exposueWindowList.AddRange(exposueWindowList);
+
+            dailySummaryList.Sort((a, b) => a.DateMillisSinceEpoch.CompareTo(b.DateMillisSinceEpoch));
+            exposueWindowList.Sort((a, b) => a.DateMillisSinceEpoch.CompareTo(b.DateMillisSinceEpoch));
+
+            await SaveExposureDataAsync(dailySummaryList, exposueWindowList);
```

Comment:
`_loggerService.EndMethod();` が呼ばれていません。

```suggestion
            await SaveExposureDataAsync(dailySummaryList, exposueWindowList);
            
            _loggerService.EndMethod();
```

Reply:
eb81000 で対応しました。
変更場所の関係で上記コードがOutdatedにならないのですが、次の箇所です。

https://github.com/cocoa-mhlw/cocoa/pull/573/commits/eb81000b47e8ddf4ac0802359a38786f5cde7176#diff-4aa40f83caacf8ecc6d54f0bd417fb05569924a4bf17e4b015e86722708cb74dR186